addons/sale_timesheet/models/sale_order.py

```diff
@@ -48,17 +48,38 @@ def _compute_field_value(self, field):
             return
 
         # Get SOs which their state is not equal to upselling or invoied and if at least a SOL has warning prepaid service upsell set to True and the warning has not already been displayed
-        filtered_self = self.filtered(lambda so: (so.user_id or so.partner_id.user_id) and so.invoice_status not in ('upselling', 'invoiced') and any(sol._has_warning_prepaid_service_upsell() and not sol.has_displayed_warning_upsell for sol in so.order_line))
-        if not filtered_self:
-            return
-        warning_upsell_sols = filtered_self.order_line._get_warning_prepaid_service_upsell_sols()
-        filtered_self = warning_upsell_sols.order_id
-        if not filtered_self:
-            return
-
-        filtered_self._create_upsell_activity()
-        # We want to display only one time the warning for each SOL
-        warning_upsell_sols.write({'has_displayed_warning_upsell': True})
+        upsellable_orders = self.filtered(lambda so:
+            so.state == 'sale'
+            and so.invoice_status not in ('upselling', 'invoiced')
+            and (so.user_id or so.partner_id.user_id)  # salesperson needed to assign upsell activity
+        )
+        for order in upsellable_orders:
+            upsellable_lines = order._filter_prepaid_service_lines_to_upsell()
+            if upsellable_lines:
+                order._create_upsell_activity()
+                # We want to display only one time the warning for each SOL
+                upsellable_lines.write({'has_displayed_warning_upsell': True})
+
+    def _filter_prepaid_service_lines_to_upsell(self):
```

Comment:
The `_filter` prefix was recommended if you kept the method on the sale.order.line level, but if it's on the order, it can stay as "_get_prepaid..." ;).
(Sorry if it was misleading :c)

Reply:
Oh okay, I understood, sorry. :)